src/ol/renderer/canvas/canvasvectortilelayerrenderer.js

```diff
@@ -0,0 +1,447 @@
+
+  if (replayContext != context) {
+    this.dispatchRenderEvent(replayContext, frameState, transform);
+    context.drawImage(replayContext.canvas, 0, 0);
+  }
+  replayContext.globalAlpha = alpha;
+
+  this.dispatchPostComposeEvent(context, frameState, transform);
+};
+
+
+/**
+ * @param {ol.VectorTile} tile Tile.
+ * @param {ol.layer.VectorTile} layer Vector tile layer.
+ * @param {number} pixelRatio Pixel ratio.
+ */
+ol.renderer.canvas.VectorTileLayer.prototype.createReplayGroup = function(tile,
+    layer, pixelRatio) {
+  var revision = layer.getRevision();
+  var renderOrder = layer.getRenderOrder();
+  if (renderOrder === undefined) {
+    renderOrder = null;
+  }
+
+  var replayState = tile.getReplayState();
+  if (!replayState.dirty && replayState.renderedRevision == revision &&
+      replayState.renderedRenderOrder == renderOrder) {
+    return;
+  }
+
+  // FIXME dispose of old replayGroup in post render
+  goog.dispose(replayState.replayGroup);
+  replayState.replayGroup = null;
+  replayState.dirty = false;
+
+  var source = layer.getSource();
+  goog.asserts.assertInstanceof(source, ol.source.VectorTile,
+      'Source is an ol.source.VectorTile');
+  var tileGrid = source.getTileGrid();
+  var tileCoord = tile.getTileCoord();
+  var pixelSpace = tile.getProjection().getUnits() == ol.proj.Units.TILE_PIXELS;
+  var extent;
+  if (pixelSpace) {
+    var tilePixelSize = source.getTilePixelSize(tileCoord[0], pixelRatio,
+        tile.getProjection());
+    extent = [0, 0, tilePixelSize[0], tilePixelSize[1]];
+  } else {
+    extent = tileGrid.getTileCoordExtent(tileCoord);
+  }
+  var resolution = tileGrid.getResolution(tileCoord[0]);
+  var tileResolution = pixelSpace ? source.getTilePixelRatio() : resolution;
+  replayState.dirty = false;
+  var replayGroup = new ol.render.canvas.ReplayGroup(0, extent,
+      tileResolution, layer.getRenderBuffer(),
+      tile.getFormat().getRightHandedPolygons());
+  var squaredTolerance = ol.renderer.vector.getSquaredTolerance(
+      tileResolution, pixelRatio);
+
+  /**
+   * @param {ol.Feature|ol.render.Feature} feature Feature.
+   * @this {ol.renderer.canvas.VectorTileLayer}
+   */
+  function renderFeature(feature) {
+    var styles;
+    if (feature.getStyleFunction()) {
+      goog.asserts.assertInstanceof(feature, ol.Feature, 'Got an ol.Feature');
+      styles = feature.getStyleFunction().call(feature, resolution);
+    } else if (layer.getStyleFunction()) {
+      styles = layer.getStyleFunction()(feature, resolution);
+    }
+    if (styles) {
+      var dirty = this.renderFeature(feature, squaredTolerance, styles,
+          replayGroup);
+      this.dirty_ = this.dirty_ || dirty;
+      replayState.dirty = replayState.dirty || dirty;
+    }
+  }
+
+  var features = tile.getFeatures();
+  if (renderOrder && renderOrder !== replayState.renderedRenderOrder) {
+    goog.array.sort(features, renderOrder);
+  }
+  goog.array.forEach(features, renderFeature, this);
```

Comment:
`features.forEach(renderFeature, this)` could be used instead of `goog.array.forEach`